transmartApp/grails-app/init/org/transmartproject/app/BootStrap.groovy

```diff
@@ -50,11 +50,6 @@ class BootStrap {
 
         // force marshaller registrar initialization
         grailsApplication.mainContext.getBean 'marshallerRegistrarService'
-
-        if ('clientCredentialsAuthenticationProvider' in
-                grailsApplication.config.grails.plugin.springsecurity.providerNames) {
-            OAuth2SyncService.syncOAuth2Clients()
-        }
     }
 
```

Comment:
According to [Grails documentation](http://docs.grails.org/latest/guide/plugins.html#providingBasicArtefacts) Bootstrap.groovy is excluded by default, when packaging a plugin. I moved the code from init() to the plugin descriptor closure: TransmartAppGrailsPlugin.doWithApplicationContext. Same for tm-oauth. There is still some config duplication in tmApp and tm-server, but I am not sure if I can safely deduplicate everything.